src/go/configgenerator/listener_generator.go

```diff
@@ -767,7 +767,7 @@ func makeBackendAuthFilter(serviceInfo *sc.ServiceInfo) *hcmpb.HttpFilter {
 			ImdsToken: &commonpb.HttpUri{
 				Uri:     fmt.Sprintf("%s%s", serviceInfo.Options.MetadataURL, util.IdentityTokenSuffix),
 				Cluster: util.MetadataServerClusterName,
-				Timeout: ptypes.DurationProto(serviceInfo.Options.HttpRequestTimeout),
+				Timeout: ptypes.DurationProto(util.DefaultImdsDeadline),
```

Comment:
how about keep the original code to use the flag --http_request_timeout_s

Reply:
Done. After discussing offline, we decided to just use the flag `--http_request_timeout_s` for all requests, including IMDS. We changed the default to `30s`.